ext/openssl/xp_ssl.c

```diff
@@ -60,9 +60,19 @@
 #define STREAM_CRYPTO_METHOD_TLSv1_2       (1<<5)
 #define STREAM_CRYPTO_METHOD_TLSv1_3       (1<<6)
 
+#ifndef OPENSSL_NO_TLS1_METHOD
+#define HAVE_TLS1 1
```

Comment:
I think you might be missing more places to make this work. From the quick look, this might need to be also covered in `php_openssl_ssl_socket_factory`. But this really needs some testing. It's also unrelated to this PR so I would prefer to have `OPENSSL_NO_TLS...` in a new PR as it might need a bit more testing.

Reply:
It's not so much about making it work as it is about making it compile. Making this work can be covered in another PR as you said.